sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1663,15 +1665,27 @@ ProgramManager::link(const std::vector<device_image_plain> &DeviceImages,
   for (const device &Dev : Devs)
     PIDevices.push_back(getSyclObjImpl(Dev)->getHandleRef());
 
+  std::vector<const char *> linkOptions;
+  for (auto &DeviceImage : DeviceImages) {
+    const std::shared_ptr<device_image_impl> &InputImpl =
+        getSyclObjImpl(DeviceImage);
+    linkOptions.push_back(InputImpl->get_bin_image_ref()->getLinkOptions());
+  }
+  std::string linkOptionsStr;
+  for (auto str : linkOptions) {
+    if (str != nullptr) {
+      linkOptionsStr += std::string(str) + " ";
```

Comment:
Can we reused applyOptionsFromImage function instead?

Reply:
I think we could. I'll try it.

Reply:
I reused applyOptionsFromImage function and remove highlited for-cycle.